recurly/resource.py

```diff
@@ -1,26 +1,36 @@
 from pydoc import locate
 
+
 class Resource:
-    @staticmethod
-    def cast(properties):
+    """Class representing a server-side object in Recurly"""
+
+    # Allows us to override resource location
+    locator = lambda class_name: locate("recurly.resources.%s" % class_name)
+
+    @classmethod
+    def cast(cls, properties):
         """Casts a dict of properties into a Recurly Resource"""
 
-        if 'object' not in properties:
+        if "object" not in properties:
             return properties
 
-        if properties['object'] == 'list' and 'data' in properties and \
-                'has_more' in properties:
-            properties['data'] = [Resource.cast(i) for i in properties['data']]
+        if (
+            properties["object"] == "list"
+            and "data" in properties
+            and "has_more" in properties
+        ):
+            properties["data"] = [Resource.cast(i) for i in properties["data"]]
             return Page(properties)
 
-        name_parts = properties['object'].split('_')
-        class_name = ''.join(x.title() for x in name_parts)
-        klass = locate("recurly.resources.%s" % class_name)
+        name_parts = properties["object"].split("_")
+        class_name = "".join(x.title() for x in name_parts)
+
+        klass = cls.locator(class_name)
 
         if klass is None:
             return properties
 
-        del properties['object']
+        del properties["object"]
```

Comment:
I know that we're basically just throwing away `properties` after this function returns but it still feels like a code smell to mutate the dict directly, rather than creating a new dict and adding just the data we need to that. I'm fine with leaving it as-is because it's a better memory tradeoff, but if we run into a weird bug because of this later I'm going to say "I told you so" lol 😉

Reply:
You're right that it's a code-smell and a good thing to consider. You don't want to do this in a public facing method. In this case, we know the function which created this object isn't doing anything with it after. I could perhaps document this method or somehow make it internal to the library. Either way, I'm not sure cloning the whole object everytime this is called would be worth it in the long run: https://github.com/recurly/recurly-client-python/blob/tooling-and-tests/recurly/base_client.py#L50